src/main/java/com/josdem/vetlog/service/impl/UserDetailsServiceImpl.java

```diff
@@ -39,14 +39,10 @@ public class UserDetailsServiceImpl implements UserDetailsService {
     @Override
     public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
         Optional<User> optional = userRepository.findByUsername(username);
-        if (optional.isPresent()) {
-            User user = optional.get();
-            return new org.springframework.security.core.userdetails.User(
-                    user.getUsername(),
-                    user.getPassword(),
-                    Arrays.asList(new SimpleGrantedAuthority(user.getRole().name())));
-        } else {
-            throw new BusinessException("User not found with username: " + username);
-        }
+        return optional.map(user -> new org.springframework.security.core.userdetails.User(
+                        user.getUsername(),
+                        user.getPassword(),
+                        Arrays.asList(new SimpleGrantedAuthority(user.getRole().name()))))
```

Comment:
I suggest using `List.of` here, the reason is that we can use an inmmutable collection here:
`List.of(new SimpleGrantedAuthority(user.getRole().name()))))`